inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/util/SchemaChangeUtils.java

```diff
@@ -0,0 +1,217 @@
+        }
+        return policyMap;
+    }
+
+    /**
+     * Serialize the policy Map[{@link SchemaChangeType}, {@link SchemaChangePolicy}] to a string
+     *
+     * @param policyMap The policy Map[{@link SchemaChangeType}, {@link SchemaChangePolicy}]
+     * @return A string format by 'key1=value1&key2=value2...'
+     */
+    public static String serialize(Map<SchemaChangeType, SchemaChangePolicy> policyMap) {
+        Preconditions.checkNotNull(policyMap, "policyMap is null");
+        StringJoiner joiner = new StringJoiner("&");
+        for (Entry<SchemaChangeType, SchemaChangePolicy> kv : policyMap.entrySet()) {
+            joiner.add(kv.getKey().getCode() + "=" + kv.getValue().getCode());
+        }
+        return joiner.toString();
+    }
+
+    /**
+     * Extract the schema change types from {@link Operation}
+     *
+     * @param operation The operation
+     * @return Set of {@link SchemaChangeType}
+     */
+    public static Set<SchemaChangeType> extractSchemaChangeTypes(Operation operation) {
+        Set<SchemaChangeType> types = new HashSet<>();
+        switch (operation.getOperationType()) {
+            case ALTER:
+                AlterOperation alterOperation = (AlterOperation) operation;
+                Preconditions.checkState(alterOperation.getAlterColumns() != null
+                        && !alterOperation.getAlterColumns().isEmpty(), "alter columns is empty");
+                for (AlterColumn alterColumn : alterOperation.getAlterColumns()) {
+                    extractSchemaChangeType(alterColumn, types);
+                }
+                break;
+            case CREATE:
+                types.add(SchemaChangeType.CREATE_TABLE);
+                break;
+            case TRUNCATE:
+                types.add(SchemaChangeType.TRUNCATE_TABLE);
+                break;
+            case RENAME:
+                types.add(SchemaChangeType.RENAME_TABLE);
+                break;
+            case DROP:
+                types.add(SchemaChangeType.DROP_TABLE);
+            default:
+        }
+        return types;
+    }
+
+    /**
+     * Extract the schema change type from {@link Operation}
+     *
+     * @param operation The operation
+     * @return A type of {@link SchemaChangeType}
+     */
+    public static SchemaChangeType extractSchemaChangeType(Operation operation) {
+        SchemaChangeType type = null;
+        switch (operation.getOperationType()) {
+            case ALTER:
+                return SchemaChangeType.ALTER;
+            case CREATE:
+                type = SchemaChangeType.CREATE_TABLE;
+                break;
+            case TRUNCATE:
+                type = SchemaChangeType.TRUNCATE_TABLE;
+                break;
+            case RENAME:
+                type = SchemaChangeType.RENAME_TABLE;
+                break;
+            case DROP:
+                type = SchemaChangeType.DROP_TABLE;
+            default:
+        }
+        return type;
```

Comment:
Maybe SchemaChangeType add a NONE enum type, default set NONE, not to return null.

Reply:
I don't think it is necessary to do this here. It is very necessary for the downstream processing based on this to let the caller know whether it really exists.

Reply:
> I don't think it is necessary to do this here. It is very necessary for the downstream processing based on this to let the caller know whether it really exists.

By NONE enum type to judge schema change type to avoid could NPE problem.